srcs/raycasting/draw_navigator.c

```diff
@@ -0,0 +1,123 @@
+#include "cub.h"
+
+static void	draw_square(t_cub *cub, int x, int y, int color)
+{
+	int i;
+	int	j;
+
+	i = 0;
+	while (i < cub->pixel_per_square)
+	{
+		j = 0;
+		while (j < cub->pixel_per_square)
+		{
+			if (i == 0 || i == cub->pixel_per_square - 1 ||
+				j == 0 || j == cub->pixel_per_square - 1)
+				my_mlx_pixel_put(cub, x + i, y + j, 0x00000000);
+			else
+				my_mlx_pixel_put(cub, x + i, y + j, color);
+			j++;
+		}
+		i++;
+	}
+}
+
+/* Borders (start/end x and y) are respective to the map grid,
+ * not to the pixels on the screen.
+ * Decided navigator cannot take more than a 1/3 of screen width/height,
+ * this value was arbitrary.
+ */
+static void	set_navigation_x_borders(t_pair_i *start, t_pair_i *end, t_cub *cub)
+{
+	int	nav_width;
+
+	nav_width = cub->pixel_per_square * cub->map_width;
+	start->x = 0;
+	end->x = cub->map_width;
+	if (nav_width > WIDTH / 3)
+	{
+		start->x = (int)(cub->pos->x - (double)WIDTH / (6.0 * (double)cub->pixel_per_square));
```

Comment:
It's working but I'm struggling a bit to understand this one:

Let's say a map width is 1000 (very big) and we're standing in pos 900 and our WIDTH is 800 and pixel per sq 10.

Then:

start->x = (int)(900 - 800 / (6 * 10) = 900 - 13.33 = 886.67 = 886
end->x = 886 + 800/(3*10) = 886 + 26.67 = 892.67 = 892 (no casting required here btw on line 42?)

Seems like a very small map as the difference between start and end x is only 6? I think I just have some things wrong in my head because it seems to run fine though. Also don't understand where the factors 3 and 6 come from.

Reply:
3 was arbitrary; 6 is 3 * 2 as I want half of the space reserved for the navigator (width / 3)
the issue there is when you are calculating end-> = 886 + 26.67 = 912.67
so the difference is actually the 26.67 which makes sense as that is 800/(3*10) so indeed 1/3 of the width divided per pixel_per_square

Reply:
Ah woops. Math 😢 . I ran some tests to further my understanding with printf and it's looking good.